### ISSUE TITLE
Improve "Upload a new dataset YAML"

### ISSUE THREAD

**Issue by malaethyca** [ethyca/fides#1531]
"Upload a new dataset YAML" allows the user to copy paste a YAML. To improve debugging:

1. Can the line number be displayed as it's done here: https://github.com/ethyca/fides/blob/main/demo_resources/demo_dataset.yml That way, when there are error messages - which tell you which line number to fix, you can actually go fix it as you can tell where you need to look based on the line numbers?
2. I copy pasted the sample dataset yaml and as you can see in the image below the qualifier lines are redlined. However, it doesn't tell you what the errors are. Worse, if you hit "Create Dataset" it goes ahead and does it anyway. So maybe it's a soft error but it's in red which is confusing.

![Image](https://user-images.githubusercontent.com/108771409/197492522-9ad94904-4df7-41b4-9354-e92efa91e5ac.png)
3. Does it make sense to have a file upload button instead of a copy paste button as copy-paste of long YAML files is error prone? This would make sense especially if as the instruction says, you would have received the file from someone rather than hand coded it right there.
4. Can we provide a link to the docs or better yet embed the docs to teach the user the syntax of the yaml, assuming they want to hand code it or in the least fix errors and need the docs to do that?

5. "Manually generate a dataset" seems permanently disabled.  Why have it then? I tried hand-coding some yaml to see if it would enable it, no luck:

![Image](https://user-images.githubusercontent.com/108771409/197493470-98368087-c1f8-4385-81e4-6e414e8edd82.png)

**Comment by mfbrown** [comment]
We could look into using the YAML editor which is used on the datastore connection UI. It is using the same library as VS Code. cc: @chriscalhoun1974

**Comment by chriscalhoun1974** [comment]
Totally agree with @mfbrown statement. Straight forward to implement and provide the user with a consistent user experience within Fides when editing Yaml markup.

**Comment by chriscalhoun1974** [comment]
@mfbrown @malaethyca Do you want to reuse the existing YAML editor when a user is creating a new or editing a connection workflow? My assumptions is yes.

![Image](https://user-images.githubusercontent.com/68459950/199539039-33a98b81-082a-407b-ae95-6b7b3bdbab62.png)
![Image](https://user-images.githubusercontent.com/68459950/199539040-a2848d57-f31a-4985-b4c0-c58e3786e846.png)

**Comment by chriscalhoun1974** [comment]
Here is my [PR](https://github.com/ethyca/fides/pull/2258).

**Comment by chriscalhoun1974** [comment]
@Kelsey-Ethyca @mfbrown How much effort should we put into the Cypress testing specifically updating the datasets.cy.ts file. Changes are needed to reflect the @monaco-editor/react NPM package. Initially, I ran into issues emulating copy/paste functionality with integrating the @monaco-editor/react NPM package. I updated the datasets.cy.ts file with TODO statements temporarily to function as a placeholder. Thoughts?
